src/main/scala/com/databricks/labs/mosaic/core/geometry/MosaicGeometryESRI.scala

```diff
@@ -29,18 +29,18 @@ abstract class MosaicGeometryESRI(geom: OGCGeometry) extends MosaicGeometry {
         }
 
     // noinspection DuplicatedCode
-    override def translate(xd: Double, yd: Double): MosaicGeometry = {
+    override def translate(xd: Double, yd: Double): MosaicGeometryESRI = {
```

Comment:
@milos-colic is this change tied to the simplification (helps with codegen) or something else?

Reply:
Yes, needed for codegen simplification, type matching doesnt work with parent classes but works with children classes.